### PR TITLE
feat: product rating

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes # https://vsf.atlassian.net/browse/SFUI2-1182

# Scope of work

product rating block 

# Screenshots of visual changes

Screenshot 2023-06-23 at 10 22 12

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: apps/docs/components/blocks/RatingForms.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+---
+# Rating Forms
+
+{{ $frontmatter.description }}
+
+## Product Rating
+
+The Product Rating block serves as a basic yet effective solution for implementing a 1-5 stars rating system commonly used in eCommerce platforms. It enables users to easily express their opinion about a particular item or experience.
+
+It allows for customization through the use of RatingButton basic component. This feature permits users to provide more specific ratings, enhancing the granularity of their feedback.
```
Comment (dkacper) [inline apps/docs/components/blocks/RatingForms.md:15]: I don't think I understand this section. I think it's about customizing stars using the RatingButton, but I don't get the other part about "granularity". Could you rephrase this section?

File: apps/docs/components/blocks/RatingForms.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+For additional customizations and more advanced functionality, you can consult our documentation on the Modal page.
```
Comment (dkacper) [inline apps/docs/components/blocks/RatingForms.md:17]: It'd be nice to link the Modal page here. I'm not sure if there's a better way that `<!-- vue --> <!-- react -->` blocks here.

File: apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          <SfButton
+            square
+            variant="tertiary"
+            className="!text-neutral-500 absolute right-2 top-2"
+            aria-label="Close product rating modal"
+            onClick={close}
+          >
+            <SfIconClose />
+          </SfButton>
+          <h3 id="modalTitle" className="font-bold typography-headline-4 md:typography-headline-3 mb-8">
```
Comment (dkacper) [inline apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx:34]: Use `useId` to generate ids, please. Same for `modalDesc` id.

```
const modalTitle = useId()
const modalDesc = useId()
```

File: apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        square
+        variant="tertiary"
+        aria-label="Close product rating modal"
+        size="sm"
+        class="!text-neutral-500 absolute right-2 top-2"
+        @click="close"
+      >
+        <SfIconClose size="sm" />
+      </SfButton>
+      <h3 id="modalTitle" class="font-bold typography-headline-4 md:typography-headline-3 mb-8">Rate your purchase</h3>
```
Comment (dkacper) [inline apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue:24]: I think we should use `useId` here as well. The only concern I have is that in case of React the hook is built-in the framework, but in Vue it's our custom composable. Nevertheless I think it's better to have IDs generated and uniq.

**Review by dkacper — COMMENTED** [review]
Blocks look very good! I left some comments on the docs and one minor improvement for IDs. Nice job! 👍🏼

File: apps/docs/components/blocks/RatingForms.md (hunk 2 above)
Comment (justyna-13) [reply apps/docs/components/blocks/RatingForms.md]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/d3797690dbd108bf5bf4029e2c4270327ee10e2a

File: apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx (hunk 3 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/d3797690dbd108bf5bf4029e2c4270327ee10e2a

File: apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue (hunk 4 above)
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/d3797690dbd108bf5bf4029e2c4270327ee10e2a

File: apps/docs/components/blocks/RatingForms.md (hunk 1 above)
Comment (justyna-13) [reply apps/docs/components/blocks/RatingForms.md:15]: As I'm not an author I've asked Jacek about clarification on jira

File: apps/docs/components/blocks/RatingForms.md (hunk 2 above)
Comment (dkacper) [inline apps/docs/components/blocks/RatingForms.md]:
```suggestion
This block is implemented using `SfRatingButton` and `SfModal` components. If you need further customizations, see their docs pages.
```
And add links to the components.

**Review by dkacper — APPROVED** [review]
Functionally looks good! Resolve the discussion around the docs how you think is going to be best, and you're free to merge. 🚀 😅

**Review by Szymon-dziewonski — APPROVED** [review]
Looking good, only minor changes from my side :) Gj

File: apps/docs/components/blocks/RatingForms.md (hunk 1 above)
Comment (justyna-13) [reply apps/docs/components/blocks/RatingForms.md]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/849575865c779fac693813b0589ad02389e9c134

File: apps/docs/components/blocks/RatingForms.md (hunk 2 above)
Comment (justyna-13) [reply apps/docs/components/blocks/RatingForms.md]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/849575865c779fac693813b0589ad02389e9c134

File: apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+          </SfButton>
+          <h3 id={modalTitle} className="font-bold typography-headline-4 md:typography-headline-3 mb-8">
+            Rate your purchase
+          </h3>
+        </header>
+        <img
+          src="http://localhost:3100/@assets/smartwatch.png"
+          alt="Smartwatch"
+          width={100}
+          height={100}
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/849575865c779fac693813b0589ad02389e9c134

File: apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue [hunk 6]
```diff
@@ -0,0 +1,49 @@
+<template>
+  <SfButton type="button" class="absolute right-1/2 top-1/2 translate-x-[50%]" @click="open"
+    >Open rating modal again</SfButton
+  >
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/849575865c779fac693813b0589ad02389e9c134

File: apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      >
+        <SfIconClose size="sm" />
+      </SfButton>
+      <h3 :id="modalTitle" class="font-bold typography-headline-4 md:typography-headline-3 mb-8">Rate your purchase</h3>
+    </header>
+    <img
+      src="http://localhost:3100/@assets/smartwatch.png"
+      alt="Smartwatch"
+      :width="100"
+      :height="100"
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/849575865c779fac693813b0589ad02389e9c134

File: apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+</template>
+
+<script lang="ts" setup>
+import { ref } from 'vue';
+import { SfModal, SfButton, SfIconClose, useDisclosure, SfRatingButton, useId } from '@storefront-ui/vue';
+
+const { isOpen, open, close } = useDisclosure({ initialValue: true });
+const modelValue = ref();
+const modalTitle = useId();
+const modalDesc = useId();
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/RatingForms/ProductRating.vue]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/849575865c779fac693813b0589ad02389e9c134

File: apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
+import { ShowcasePageLayout } from '../../showcases';
+// #region source
+import { useState, useId } from 'react';
+import { SfModal, SfButton, SfIconClose, useDisclosure, SfRatingButton } from '@storefront-ui/react';
+
+export default function ProductRating() {
+  const { isOpen, open, close } = useDisclosure({ initialValue: true });
+  const [rating, setRating] = useState(0);
+  const modalTitle = useId();
+  const modalDesc = useId();
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/next/pages/showcases/RatingForms/ProductRating.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2849/commits/849575865c779fac693813b0589ad02389e9c134
